test/nn/test_pooling.py

```diff
@@ -376,6 +376,40 @@ def test_adaptive_pooling_zero_batch(self, dtype, device):
         mod = torch.nn.AdaptiveAvgPool3d((5, 5, 5)).to(device)
         _test_module_empty_input(self, mod, inp, check_size=False)
 
+    @onlyNativeDeviceTypes
+    @dtypes(torch.float32, torch.float64)
+    @dtypesIfCUDA(torch.float32, torch.float64, torch.bfloat16, torch.float16)
+    def test_adaptive_pooling_empty_output_size(self, dtype, device):
+        error_msg = "Expected grad_output to have non-zero size for non-batch dimensions"
+
+        input = torch.rand([1, 64, 10, 9], dtype=dtype, device=device, requires_grad=True)
+        output_size = 0
+
+        for dim, pool_type in itertools.product((2, 3), ('avg', 'max')):
```

Comment:
I recommend just iterating over the actual modules or functions the test cares about instead of dynamically acquiring them by name

Reply:
Sure. I updated using actual modules rather than names. Thanks!